File: THCS/2022_2023/PhuYen_2223/Bai4/BAI4.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

int main() {
    // Đọc dữ liệu từ file BAI4.INP
    ifstream inp_file("BAI4.INP");
    int N, K;
    inp_file >> N >> K;
    
    vector<int> scores(N);
    for (int i = 0; i < N; i++) {
        inp_file >> scores[i];
    }
    inp_file.close();
    
    // Biến lưu tổng lớn nhất và chỉ số bắt đầu của đội
    int max_sum = -1;
    int start_index = 0;

    // Tính tổng của đội gồm K bạn liên tiếp
    int current_sum = 0;
    for (int i = 0; i < K; i++) {
        current_sum += scores[i];
    }
    max_sum = current_sum;
    start_index = 0;
    
    // Duyệt qua các đoạn con K phần tử
    for (int i = K; i < N; i++) {
        current_sum = current_sum - scores[i - K] + scores[i];  // Trượt cửa sổ
        if (current_sum > max_sum || (current_sum == max_sum && i > start_index + K - 1)) {
            max_sum = current_sum;
            start_index = i - K + 1;
        }
    }
    
    // Ghi kết quả vào file BAI4.OUT
    ofstream out_file("BAI4.OUT");
    out_file << max_sum << endl;
    for (int i = start_index; i < start_index + K; i++) {
        out_file << (i + 1) << " ";  // Số báo danh bắt đầu từ 1
    }
    out_file << endl;
    out_file.close();

    return 0;
}

File: THCS/2022_2023/PhuYen_2223/Bai4/BAI4.py
def find_best_team(N, K, scores):
    # Biến lưu tổng lớn nhất và chỉ số bắt đầu của đội
    max_sum = -1
    start_index = 0

    # Tính tổng của đội gồm K bạn liên tiếp đầu tiên
    current_sum = sum(scores[:K])
    max_sum = current_sum
    start_index = 0

    # Duyệt qua các đoạn con K phần tử tiếp theo
    for i in range(K, N):
        current_sum = current_sum - scores[i - K] + scores[i]
        if current_sum > max_sum or (current_sum == max_sum and i > start_index + K - 1):
            max_sum = current_sum
            start_index = i - K + 1

    # Trả về tổng lớn nhất và đội gồm K bạn
    return max_sum, [i + 1 for i in range(start_index, start_index + K)]

# Đọc dữ liệu từ file BAI4.INP
with open('BAI4.INP', 'r') as inp_file:
    # Dòng đầu tiên: N và K
    N, K = map(int, inp_file.readline().split())

    # Dòng thứ hai: số câu trả lời đúng của từng bạn đoàn viên
    scores = list(map(int, inp_file.readline().split()))

# Tìm đội có tổng số câu trả lời đúng nhiều nhất
max_sum, best_team = find_best_team(N, K, scores)

# Ghi kết quả vào file BAI4.OUT
with open('BAI4.OUT', 'w') as out_file:
    out_file.write(f"{max_sum}\n")
    out_file.write(" ".join(map(str, best_team)) + "\n")
